include/cosim/osp_config_parser.hpp

```diff
@@ -27,6 +27,9 @@ struct osp_config
     /// The default start time for a simulation
     time_point start_time;
 
+    /// The (optional) end time for a simulation
+    time_point end_time;
```

Comment:
Should this be `std::optional<time_point>`, or are you using zero to indicate "not specified"?

Reply:
Changed to `std::optional` here. 

I'm not really into where to use and not use `std::optional` in APIs, but I think it makes sense here since it is optional in the XML. I kept it as a `double` in the `SimulationInformation` struct.